source/CModuleSystem.cpp

```diff
@@ -142,11 +127,73 @@ void CLEO::CModuleSystem::NormalizePath(std::string& path)
 	};
 }
 
+void CModuleSystem::CModule::Update()
+{
+	while (updateActive)
+	{
+		if (!updateNeeded)
+		{
+			std::filesystem::file_time_type time;
+			try
+			{
+				time = std::filesystem::last_write_time(filepath);
+			}
+			catch (...)
+			{
+				time = {};
+			}
+
+			// file not exists or up to date
+			if (time == std::filesystem::file_time_type{} || time == fileTime)
+			{
+				// query files once a second
+				for(size_t i = 0; i < 100 && updateActive; i++)
+					std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
why don't we sleep for 1000 ms once instead?

Reply:
also it could be a better idea to use some well tested library.
https://github.com/SpartanJ/efsw
https://github.com/ThomasMonkman/filewatch

Reply:
It won't check for updateActive active then, which will result in 1s delay for each module destructor call.

Reply:
I guess it's fine? 1s to switch between compiler & game window. CLEO Redux uses 1s sleep

Reply:
> also it could be a better idea to use some well tested library. https://github.com/SpartanJ/efsw https://github.com/ThomasMonkman/filewatch

I had look at the library you linked earlier, but having global watcher sounded problematic in terms of threads synchronization etc. 
Module would need it's thread to load the file anyway. Std:: is "well tested" library too. Threads and file system features provided by newer cpp versions are cross platform too.